### DISCUSSION TITLE
Usage with echo framework and cookie auth

### DISCUSSION THREAD

**Discussion by TheLoneDeveloper** [zishang520/socket.io#36] [Q&A]
Hello just wanted to ask if this is possible with this package

I have an echo framework app that I’m adding realtime functions to and we use cookie based auth. Is it possible to read the session cookie inside io.On(“connection”) and save the user id to the socket data somehow? I have tried going through the package docs but I can’t see how you would get the original request like you can in the original JavaScript version of this package for nodejs?

**Comment by zishang520** [comment]
The following code snippet is for your reference:
```go
	httpServer := types.NewWebServer(nil)
	socketio := socket.NewServer(httpServer, nil)
	socketio.On("connection", func(clients ...interface{}) {
		client := clients[0].(*socket.Socket)
		utils.Log().Success("cookies: %v", client.Client().Request().Request().Cookies())
	})
```

**Reply by TheLoneDeveloper** [reply]
Is it possible to use echo frameworks session middleware for this then or access the echo context inside the handshake?

**Reply by zishang520** [reply]
I haven't studied the echo framework, so I can't give you an answer.
